base/platform/linux/base_cellular_utilities_linux.cpp

```diff
@@ -0,0 +1,159 @@
+// This file is part of Desktop App Toolkit,
+// a set of libraries for developing nice desktop applications.
+//
+// For license and copyright information please follow this link:
+// https://github.com/desktop-app/legal/blob/master/LEGAL
+//
+#include "base/platform/linux/base_cellular_utilities_linux.h"
+
+#include "base/const_string.h"
+
+#ifndef DESKTOP_APP_DISABLE_DBUS_INTEGRATION
+#include "base/platform/linux/base_linux_glibmm_helper.h"
+
+#include <glibmm.h>
+#include <giomm.h>
+#endif // !DESKTOP_APP_DISABLE_DBUS_INTEGRATION
+
+namespace base::Platform {
+namespace {
+
+constexpr auto kMMService = "org.freedesktop.ModemManager1"_cs;
+
+typedef enum {
+	MM_MODEM_MODE_NONE = 0,
+	MM_MODEM_MODE_CS   = 1 << 0,
+	MM_MODEM_MODE_2G   = 1 << 1,
+	MM_MODEM_MODE_3G   = 1 << 2,
+	MM_MODEM_MODE_4G   = 1 << 3,
+	MM_MODEM_MODE_5G   = 1 << 4,
+	MM_MODEM_MODE_ANY  = 0xFFFFFFFF
+} MMModemMode;
+
+typedef enum {
+	MM_MODEM_LOCATION_SOURCE_NONE          = 0,
+	MM_MODEM_LOCATION_SOURCE_3GPP_LAC_CI   = 1 << 0,
+	MM_MODEM_LOCATION_SOURCE_GPS_RAW       = 1 << 1,
+	MM_MODEM_LOCATION_SOURCE_GPS_NMEA      = 1 << 2,
+	MM_MODEM_LOCATION_SOURCE_CDMA_BS       = 1 << 3,
+	MM_MODEM_LOCATION_SOURCE_GPS_UNMANAGED = 1 << 4,
+	MM_MODEM_LOCATION_SOURCE_AGPS_MSA      = 1 << 5,
+	MM_MODEM_LOCATION_SOURCE_AGPS_MSB      = 1 << 6,
+} MMModemLocationSource;
+
+#ifndef DESKTOP_APP_DISABLE_DBUS_INTEGRATION
+std::vector<CellNetwork> MMCellNetworks() {
+	std::vector<CellNetwork> result;
+
+	try {
```

Comment:
Almost the whole MMCellNetworks function consists of try...catch. You can remove extra braces near the function definition.

```c++
std::vector<CellNetwork> MMCellNetworks()
try {
    std::vector<CellNetwork> result;
    const auto connection = ...
    
    return result;
} catch (...) {
    DEBUG_LOG("Something went wrong");
    return {};
}
```